### PR TITLE
New README proposal

### PR DESCRIPTION
**Pull request by LilianBoulard**
As discussed in #272, this PR proposes [a new README for the project](https://github.com/LilianBoulard/dirty_cat/tree/new_readme_proposal).
It also adds/updates some information on the website, mainly in the index.

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
I worry that this duplicates too much information from the website. Duplicated information will be hard to maintain and will fall out of sync

**Comment by jovan-stojanovic** [comment]
Apart from the "When is it useful?" part which is too detailed, I find you it's great as it's much clearer.
Maybe you can remove/shorten and merge the "When is it useful?" and "What dirty_cat does not"?
In any case, you put the link to the website every time, and for every encoder, people will tend to click on it for more information.

File: README.rst [hunk 1]
```diff
@@ -6,57 +6,138 @@ dirty_cat
...
 .. |circleci| image:: https://img.shields.io/circleci/build/github/dirty-cat/dirty_cat/master?label=CircleCI
-.. |Black| image:: https://img.shields.io/badge/code%20style-black-000000.svg
+.. |black| image:: https://img.shields.io/badge/code%20style-black-000000.svg
 
-dirty_cat is a Python module for machine-learning on dirty categorical variables.
+`dirty_cat <https://dirty-cat.github.io/>`_ is a Python library
+that facilitates machine-learning on dirty categorical variables.
 
-Website: https://dirty-cat.github.io/
+The techniques implemented are based on science!
```
Comment (GaelVaroquaux) [inline README.rst:21]:
```suggestion
```

File: README.rst [hunk 2]
```diff
@@ -6,57 +6,138 @@ dirty_cat
...
+For a detailed description of the problem of encoding dirty categorical data, see
+`Similarity encoding for learning with dirty categorical variables <https://hal.inria.fr/hal-01806175>`_ [1]_
+and `Encoding high-cardinality string categorical variables <https://hal.inria.fr/hal-02171256v4>`_ [2]_.
```
Comment (GaelVaroquaux) [inline README.rst:24]:
```suggestion
For a detailed description of the problem of encoding dirty categorical data, see `the website <https://dirty-cat.github.io/>`_```

File: README.rst [hunk 3]
```diff
@@ -6,57 +6,138 @@ dirty_cat
...
-dirty_cat requires:
+Dependencies and minimal versions are listed in the file `min-requirements.txt <https://github.com/dirty-cat/dirty_cat/blob/master/requirements-min.txt>`_
 
-- Python (>= 3.8)
-- NumPy (>= 1.17.3)
-- SciPy (>= 1.4.0)
-- scikit-learn (>= 0.22.0)
-- pandas (>= 1.2.0)
+Related projects
+----------------
```
Comment (GaelVaroquaux) [inline README.rst:104]: This information should be added somewhere on the website, and not on the README. The README is not the documentation. It is less visible, and less searchable than a web page.

**Review by GaelVaroquaux — COMMENTED** [review]
I am not commenting on all the details of this PR, but in general I find that it puts way too much information in the readme. This information should be put on the web page, where it can be laid out in a visible and searchable way. Adding it here runs the risk of duplicated information that is not kept up to date.

**Comment by LilianBoulard** [comment]
Okay, since these changes are a point of contention, we should delay the PR, so we can release on Monday, and we can discuss it together on Wednesday, during the sprint :)

**Comment by LilianBoulard** [comment]
Okay, I've moved some parts on the website, which I agree is more suitable.
Please check it out and let me know what you think!

**Comment by jovan-stojanovic** [comment]
I like what you added to the website! 
For the part `What can dirty_cat do`, I fear it will have to change often. We have added recently `fuzzy_join` and we will add soon `deduplication`, so it will evolve fast. 
The rest are all good improvements to me.

**Review by jovan-stojanovic — APPROVED** [review]

**Comment by LeoGrin** [comment]
Great! Maybe it would be cool to add one line on the SuperVectorizer in "What can dirty-cat do?" ? It doesnt' really fit in the rest of the description and it's a really cool feature.

**Comment by LilianBoulard** [comment]
I get what you mean, but I feel like this is true for both the SuperVectorizer and the fuzzy join. Without completely reverting what I did earlier, I added some very brief mentions of the most important tools and encoders, in order to guide users.

**Comment by LeoGrin** [comment]
LGTM

**Comment by LilianBoulard** [comment]
Alright, thanks all for the reviews, merging!

**Pull request merged**